src/libraries/System.Private.Xml.Linq/tests/xNodeReader/XNodeReaderAttributeTests.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections.Generic;
+using CoreXml.Test.XLinq;
+using Xunit;
+
+namespace System.Xml.Linq.xNodeReader.Tests
+{
+    public class XNodeReaderAttributeTests
+    {
+        [Fact]
+        public void GetAttributeThrowsOnIndexMinusOne()
+        {
+            XmlReader dataReader = GetReaderFromXDocumentAndPositionOnElement();
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => dataReader.GetAttribute(-1));
+        }
+
+        [Fact]
+        public void GetAttributeThrowsOnIndexMinusTwo()
+        {
+            XmlReader dataReader = GetReaderFromXDocumentAndPositionOnElement();
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => dataReader.GetAttribute(-2));
+        }
+
+        [Fact]
+        public void IndexerThrowsOnIndexMinusOne()
+        {
+            XmlReader dataReader = GetReaderFromXDocumentAndPositionOnElement();
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => dataReader[-1]);
+        }
+
+        [Fact]
+        public void IndexerThrowsOnIndexMinusTwo()
+        {
+            XmlReader dataReader = GetReaderFromXDocumentAndPositionOnElement();
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => dataReader[-2]);
+        }
+
+        [Fact]
+        public void IndexerThrowsOnNegativeIndicesOnXNodes()
+        {
+            foreach (XNode n in GetXNodeTypes())
+            {
+                using XmlReader r = n.CreateReader();
+
+                r.Read();
+
+                Assert.Throws<ArgumentOutOfRangeException>(() => r[-100000]);
+                Assert.Throws<ArgumentOutOfRangeException>(() => r[-1]);
+            }
+        }
+
+        [Fact]
+        public void GetAttributeThrowsOnNegativeIndicesOnXNodes()
+        {
+            foreach (XNode n in GetXNodeTypes())
+            {
+                using XmlReader r = n.CreateReader();
+
+                r.Read();
+
+                Assert.Throws<ArgumentOutOfRangeException>(() => r.GetAttribute(-100000));
+                Assert.Throws<ArgumentOutOfRangeException>(() => r.GetAttribute(-1));
+            }
+        }
+
+        [Fact]
+        public void GetAttributeIntThrowsInNonInteractiveMode()
+        {
+            foreach (XNode n in GetXNodeTypes())
+            {
+                using XmlReader r = n.CreateReader();
+
+                Assert.Throws<InvalidOperationException>(() => r.GetAttribute(0));
+                Assert.Throws<InvalidOperationException>(() => r.GetAttribute(100000));
+            }
+        }
+
+        [Fact]
+        public void GetAttributeStringReturnsNullInNonInteractiveMode()
+        {
+            foreach (XNode n in GetXNodeTypes())
+            {
+                using XmlReader r = n.CreateReader();
+
+                Assert.Null(r.GetAttribute(null));
+                Assert.Null(r.GetAttribute(null, null));
+                Assert.Null(r.GetAttribute(""));
+                Assert.Null(r.GetAttribute("", ""));
+            }
+        }
+
+        [Fact]
```

Comment:
can you add a test case where XML element has exactly one attribute and you read index=1?

Reply:
Done.